src/components/Form/index.tsx

```diff
@@ -42,9 +42,9 @@ export interface FormValidationError {
 export interface FormProps<FormValues> extends StandardProps {
   /**
    * Shows the given message if the user wants to navigate
-   * with changes being made.
+   * with changes being made or renders custom component with message if provided.
    */
-  prompt?: string;
+  prompt?: (changed: boolean) => React.ReactChild | string;
```

Comment:
The example with a simple prompt is no longer working for a Form component.
```ts
<Form prompt="Are you sure?">
 <TextField/>
 <Button>Ok</Button>
</Form>
```

I didn't dig into that to check the reason. Could you verify it?

PS: If it's related to necessary to include `react-router` to make it work within a form then it is a breaking change.

Reply:
good catch. The general page reload, or leaving the app stopped working. The reason, I forgot that it is supported regardless, if there is react-router or not. I will update my PR